drivers/display/display_st7701.c

```diff
@@ -0,0 +1,446 @@
+	} else if (cfg->rotation == 270) {
+		data->xres = cfg->height;
+		data->yres = cfg->width;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_270;
+	}
+
+	/* attach to MIPI-DSI host */
+	mdev.data_lanes = cfg->data_lanes;
+	mdev.pixfmt = data->dsi_pixel_format;
+	mdev.mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM;
+
+	mdev.timings.hactive = cfg->width;
+	mdev.timings.hbp = 40;
+	mdev.timings.hsync = 32;
+	mdev.timings.hfp = 8;
+	mdev.timings.vactive = cfg->height;
+	mdev.timings.vbp = 6;
+	mdev.timings.vsync = 8;
+	mdev.timings.vfp = 9;
+
+	ret = mipi_dsi_attach(cfg->mipi_dsi, cfg->channel, &mdev);
+	if (ret < 0) {
+		LOG_ERR("MIPI-DSI attach failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = st7701_check_id(dev);
+	if (ret) {
+		LOG_ERR("Panel ID check failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = st7701_configure(dev);
+	if (ret) {
+		LOG_ERR("DSI init sequence failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = st7701_blanking_off(dev);
+	if (ret) {
+		LOG_ERR("Display blanking off failed! (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+#define ST7701_DEVICE(inst)                                                                        \
+	static const struct st7701_config st7701_config_##inst = {                                 \
+		.mipi_dsi = DEVICE_DT_GET(DT_INST_BUS(inst)),                                      \
+		.reset = GPIO_DT_SPEC_INST_GET_OR(inst, reset_gpios, {0}),                         \
+		.backlight = GPIO_DT_SPEC_INST_GET_OR(inst, bl_gpios, {0}),                        \
+		.data_lanes = DT_INST_PROP_BY_IDX(inst, data_lanes, 0),                            \
+		.width = DT_INST_PROP(inst, width),                                                \
+		.height = DT_INST_PROP(inst, height),                                              \
+		.channel = DT_INST_REG_ADDR(inst),                                                 \
+		.rotation = DT_INST_PROP(inst, rotation),                                          \
+		.gip_e0 = DT_INST_PROP(inst, gip_e0),                                              \
+		.gip_e1 = DT_INST_PROP(inst, gip_e1),                                              \
+		.gip_e2 = DT_INST_PROP(inst, gip_e2),                                              \
+		.gip_e3 = DT_INST_PROP(inst, gip_e3),                                              \
+		.gip_e4 = DT_INST_PROP(inst, gip_e4),                                              \
+		.gip_e5 = DT_INST_PROP(inst, gip_e5),                                              \
+		.gip_e6 = DT_INST_PROP(inst, gip_e6),                                              \
+		.gip_e7 = DT_INST_PROP(inst, gip_e7),                                              \
+		.gip_e8 = DT_INST_PROP(inst, gip_e8),                                              \
+		.gip_eb = DT_INST_PROP(inst, gip_eb),                                              \
+		.gip_ec = DT_INST_PROP(inst, gip_ec),                                              \
```

Comment:
Since the properties are not required by the yaml file
```suggestion
		.gip_e0 = DT_INST_PROP_OR(inst, gip_e0, {}),                                              \
		.gip_e1 = DT_INST_PROP_OR(inst, gip_e1, {}),                                              \
		.gip_e2 = DT_INST_PROP_OR(inst, gip_e2, {}),                                              \
		.gip_e3 = DT_INST_PROP_OR(inst, gip_e3, {}),                                              \
		.gip_e4 = DT_INST_PROP_OR(inst, gip_e4, {}),                                              \
		.gip_e5 = DT_INST_PROP_OR(inst, gip_e5, {}),                                              \
		.gip_e6 = DT_INST_PROP_OR(inst, gip_e6, {}),                                              \
		.gip_e7 = DT_INST_PROP_OR(inst, gip_e7, {}),                                              \
		.gip_e8 = DT_INST_PROP_OR(inst, gip_e8, {}),                                              \
		.gip_eb = DT_INST_PROP_OR(inst, gip_eb, {}),                                              \
		.gip_ec = DT_INST_PROP_OR(inst, gip_ec, {}),                                              \
```